tools/c7n_azure/c7n_azure/resources/vm.py

```diff
@@ -23,4 +24,21 @@ class resource_type(object):
         client = 'ComputeManagementClient'
         enum_spec = ('virtual_machines', 'list_all')
         id = 'id'
-        name = 'name'
\ No newline at end of file
+        name = 'name'
+        default_report_fields = (
+            'name',
+            'location',
+            'properties.hardwareProfile.vmSize',
+        )
+
+@VirtualMachine.filter_registry.register('instance-view')
+class InstanceViewFilter(ValueFilter):
+    schema = type_schema('instance-view', rinherit=ValueFilter.schema)
+
```

Comment:
so we've been annotating IAM permissions typically for filters and actions, so that we can gen access/rbac policies for custodian execution based on a user's given policies (ie minimal permission set based on things to do), i'm not familiar with azure iam/rbac models but something to keep in mind.

Reply:
I saw that was happening and haven't had a chance to work through what it means for us.   Filed #2273